File: src/main/java/com/github/kindrat/cassandra/client/filter/fields/RowPredicate.java
package com.github.kindrat.cassandra.client.filter.fields;

import com.datastax.driver.core.CodecRegistry;
import com.datastax.driver.core.TypeCodec;
import com.github.kindrat.cassandra.client.ui.DataObject;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.function.Predicate;

import static lombok.AccessLevel.PROTECTED;

@Getter(PROTECTED)
@RequiredArgsConstructor
public abstract class RowPredicate implements Predicate<DataObject> {
    @Getter(PROTECTED)
    private static final CodecRegistry codecRegistry = new CodecRegistry();
    private final String field;
    private final String value;

    TypeCodec<?> getColumnCodec(Object actual) {
        return getCodecRegistry().codecFor(actual);
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/filter/fields/RowPredicate.kt
package com.github.kindrat.cassandra.client.filter.fields

import com.datastax.oss.driver.api.core.type.codec.TypeCodec
import com.datastax.oss.driver.api.core.type.codec.registry.CodecRegistry
import com.github.kindrat.cassandra.client.ui.DataObject
import java.util.function.Predicate

abstract class RowPredicate(val field: String, val value: String) : Predicate<DataObject> {

    companion object {
        private val codecRegistry = CodecRegistry.DEFAULT

        fun getColumnCodec(actual: Any): TypeCodec<*> {
            return codecRegistry.codecFor(actual)
        }
    }
}